retrofit/src/main/java/retrofit/RequestFactoryParser.java

```diff
@@ -405,15 +405,31 @@ private void parseParameters(Retrofit retrofit) {
             com.squareup.okhttp.Headers headers = com.squareup.okhttp.Headers.of(
                 "Content-Disposition", "form-data; name=\"" + part.value() + "\"",
                 "Content-Transfer-Encoding", part.encoding());
-            Converter<?, RequestBody> converter;
-            try {
-              converter =
+
+            Class<?> rawParameterType = Utils.getRawType(methodParameterType);
+            if (Iterable.class.isAssignableFrom(rawParameterType)) {
+              if (!(methodParameterType instanceof ParameterizedType)) {
```

Comment:
Deduped in https://github.com/iwo/retrofit/commit/a58dd5985a625569f10bcaec53581ad81c4b0212